src/Microsoft.AspNet.Mvc.Core/Controller.cs

```diff
@@ -1036,6 +1036,94 @@ public virtual Task<bool> TryUpdateModelAsync<TModel>([NotNull] TModel model)
         }
 
         /// <summary>
+        /// Updates the specified <paramref name="model"/> instance using values from the controller's current
+        /// <see cref="IValueProvider"/> and a <paramref name="prefix"/>.
+        /// </summary>
+        /// <param name="model">The model instance to update.</param>
+        /// <param name="modelType">The type of model instance to update.</param>
+        /// <param name="prefix">The prefix to use when looking up values in the current <see cref="IValueProvider"/>
+        /// </param>
+        /// <returns>A <see cref="Task"/> that on completion returns <c>true</c> if the update is successful</returns>
+        [NonAction]
+        public virtual async Task<bool> TryUpdateModelAsync([NotNull] object model,
+                                                            [NotNull] Type modelType,
+                                                            [NotNull] string prefix)
+        {
+            if (!modelType.IsAssignableFrom(model.GetType()))
```

Comment:
should this code be in one common place ( ```ModelBindingHelper.TryUpdateModel``` )? would you not need to validate this for ModelBindingHelper.TryUpdateModel.